components/clp-py-utils/pyproject.toml

```diff
@@ -16,6 +16,7 @@ boto3 = "^1.35.81"
 mariadb = "~1.0.11"
 mysql-connector-python = "^8.2.0"
 pydantic = "^2.11.9"
+pyinstrument = "^5.0.0"
```

Comment:
do we plan to include the profiling feature in the formal releases? if not, maybe we should put this into a
```
[dependency-groups]
dev = [
    ...
]
```

that said, the `pyinstrument` package is only ~270KB (https://pypi.org/project/pyinstrument/#files) so it shouldn't hurt to be bundled into the release.